### PR TITLE
Issue 2999: Better Mastodon handling

### PR DESCRIPTION
**Pull request by annando**
This is a partly handling of issue https://github.com/friendica/friendica/issues/2999 for Mastodon. The Blablanet handling is still open.

### CODE REVIEW

**Comment by annando** [comment]
BTW: This also fixes issue https://github.com/friendica/friendica/issues/2963

**Comment by tobiasd** [comment]
These other three issues can all be closed?

**Comment by annando** [comment]
I don't know what causes the missing avatar picture in issue https://github.com/friendica/friendica/issues/2957 but the rest should work.

File: include/socgraph.php [hunk 1]
```diff
@@ -755,6 +764,13 @@ function poco_check_server($server_url, $network = "", $force = false) {
 						$versionparts = explode("-", $version);
 						$version = $versionparts[0];
 					}
+
+					if(stristr($line,'Server: Mastodon')) {
```
Comment (Hypolite) [inline include/socgraph.php]: Standars: please add a space between `if` and the opening parentheses.

**Comment by tobiasd** [comment]
@Hypolite can you have a look if your issues with Mastodon are solved now and eventually close them?

**Comment by Hypolite** [comment]
I'll start reviewing them tonight EST. Most of them involve someone posting something on Mastodon, so we will have to wait a bit.

**Pull request merged**